sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/MutableColumnarRow.java

```diff
@@ -146,9 +146,7 @@ public UTF8String getUTF8String(int ordinal) {
   @Override
   public CalendarInterval getInterval(int ordinal) {
     if (columns[ordinal].isNullAt(rowId)) return null;
```

Comment:
Do we still need this null check?

Reply:
see https://github.com/apache/spark/pull/20438#discussion_r164719793 . In this PR I just fixed the returning null issue for `getStruct` and `getInterval`, because they are non-abstract. There should be a follow up to clearly document that `ColumnVector.getBinary/getUTF8String/...` should return null if this slot is null. Then we can remove these null checks here. I appreciate it if you have time to take this :)

Reply:
Let me try to prepare a PR tonight.